tools/ipapy2/ipapy2_compute_bio.xml

```diff
@@ -0,0 +1,67 @@
+<tool id="ipapy2_compute_bio" name="IPA compute bio" version="@TOOL_VERSION@+galaxy0" profile="21.05">
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">ipapy2</requirement>
+    </requirements>
+
+    <command detect_errors="exit_code"><![CDATA[
+        python3 '${__tool_directory__}/ipapy2_compute_bio.py'
+        --MS1_DB '${MS1_DB}'
+        --annotations '${annotations}'
+        --biochemical_mode '${biochemical_mode.biochemical_mode}'
+        #if $biochemical_mode.biochemical_mode == "connections"
+            --connection_list '${biochemical_mode.connection_list}'
+        #end if
+        --compute_bio_output "${compute_bio_output}"
+    ]]></command>
+
+    <inputs>
+        <param label="MS1 DB table" name="MS1_DB" type="data" format="csv" help="pandas dataframe containing the MS1 database."/>
+        <param label="annotations" name="annotations" type="data" format="csv" optional="true" help="pandas dataframe containing all the possible annotations for the measured features."/>
+        <conditional name="biochemical_mode">
+            <param name="biochemical_mode" type="select" label="biochemical mode"
+                    help="connections are computed based on the reactions or connections.">
+                    <option value="reactions" selected="true">reactions</option>
+                    <option value="connections">connections</option>
+            </param>
+            <when value="reactions">
+            </when>
+            <when value="connections">
+                <param name="connection_list" type="text" optional="true" label="connections list provided" 
+                        help="list of possible connections between compounds defined as formulas. Only necessary if mode='connections'."/>           
+            </when>
+        </conditional>
+    </inputs>
```

Comment:
How does this work if the value is selected to be reactions?

Reply:
It falls back on the default connections passed in the code:

https://github.com/francescodc87/ipaPy2/blob/bebd9126428bba9e64845e057df4892dd7c88b7c/ipaPy2/ipa.py#L659